python/src/aiconfig/editor/client/src/components/prompt/PromptContainer.tsx

```diff
@@ -66,10 +70,54 @@ export default memo(function PromptContainer({
     [promptId, onUpdateParameters]
   );
 
-  const runPrompt = useCallback(
-    async () => await onRunPrompt(promptId),
-    [promptId, onRunPrompt]
+  const [missingRequiredFields, setMissingRequiredFields] = useState(
+    new Set<string>()
   );
+  const onUpdateMissingRequiredFields = useCallback(
+    (fieldName: string, fieldValue: JSONValue) => {
+      setMissingRequiredFields((prevState) => {
+        const newState = new Set<string>(prevState);
+        const isPrimitiveValue =
+          typeof fieldValue === "string" ||
+          typeof fieldValue === "number" ||
+          typeof fieldValue === "boolean" ||
+          typeof fieldValue === "undefined" ||
+          typeof fieldValue === "bigint";
```

Comment:
I realized a lot of the time property from `SettingsPropertyRenderer` was an object, and I finally realized it's basically just recursively checking it's own JSON file at each nested level. But yea added this check here so that we don't do any operations on values that don't make sense